src/coreclr/jit/hwintrinsic.h

```diff
@@ -895,6 +895,14 @@ struct HWIntrinsicInfo
     }
 
 #ifdef TARGET_ARM64
+    static bool HasImmediateOperandRange(NamedIntrinsic id)
```

Comment:
`DuplicateSelectedScalarToVector128` has `HW_Flag_HasImmediateOperand` and not any of the `HW_Category_` flag in this method, but still it should check for the range.

```c#
/// <summary>
        /// uint8x16_t vdupq_lane_u8 (uint8x8_t vec, const int lane)
        ///   A32: VDUP.8 Qd, Dm[index]
        ///   A64: DUP Vd.16B, Vn.B[index]
        /// </summary>
        public static [Vector128](https://source.dot.net/System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector128_1.cs.html#a7d647b6968ad0f9)<byte> [DuplicateSelectedScalarToVector128](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/AdvSimd.PlatformNotSupported.cs,c98f4003aa811a2f)([Vector64](https://source.dot.net/System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector64_1.cs.html#576983e071d8f012)<byte> value, [[ConstantExpected](https://source.dot.net/System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/ConstantExpectedAttribute.cs.html#12103fea5f68fa88)([Max](https://source.dot.net/System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/ConstantExpectedAttribute.cs.html#bcd0dee62624224e) = (byte)(7))] byte index) { throw new [PlatformNotSupportedException](https://source.dot.net/System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/PlatformNotSupportedException.cs.html#ad07c3864de72561)(); }
```

Reply:
It should, though I'm not sure the best way to cover all the intrinsics that have a possible ArgumentOutOfRangeException without going through them one by one (which would take some time). At least this gets most of it blocked and we can fix the others as they come in through Fuzzlyn.

Reply:
can you open an issue for this?